pkg/version/version.go

```diff
@@ -34,26 +33,32 @@ type CiliumVersion struct {
 	GoRuntimeVersion string
 	// Arch is the architecture where Cilium was compiled
 	Arch string
+	// BuildDate is the build time reference stored as string
+	BuildDate string
 }
 
+// Version is set during build
 var Version string
 
-func versionFrom(versionString string) CiliumVersion {
-	cver := CiliumVersion{}
-	output := strings.Replace(versionString, "Cilium ", "", 1)
-
-	fields := strings.Split(output, " ")
-	cver.Version = fields[0]
-	cver.Revision = fields[1]
-	cver.Arch = fields[len(fields)-1]
-	cver.GoRuntimeVersion = runtime.Version()
+// FromString converts a version string into struct
+func FromString(versionString string) CiliumVersion {
+	// string to parse: "0.11.90 774ecd3 Wed, 19 Jul 2017 06:27:28 +0000 go version go1.8.3 linux/amd64"
+	regex := regexp.MustCompile(`^(\d+)\.(\d+)\.(\d+)\s(\w{7})\s(\w{3},\s\d{1,2}\s\w{3}\s\d{4}\s\d{2}:\d{2}:\d{2}\s[+-]\d{4})\sgo\sversion\s(go\d[\.\d]+)\s(\w+\/\w+)$`)
```

Comment:
If parsing the build date+time is so tricky, I'd refactor it to use the ISO 8601 / RFC 3339 format (with "T" as separator).